backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -64,6 +65,21 @@ export class FeatureFlagService {
 
     const includedFeatureFlags = await this.featureFlagLevelInclusionExclusion(filteredFeatureFlags, experimentUserDoc);
 
+    // save exposures in db
+    try {
+      const exposureRepo = getRepository(FeatureFlagExposure);
+
+      await Promise.all(
```

Comment:
Instead of using a `map` which will do multiple DB requests, you should use an array to save all the documents in a single request.